src/main/kotlin/org/dashj/platform/dpp/statetransition/StateTransitionIdentitySigned.kt

```diff
@@ -137,28 +152,43 @@ abstract class StateTransitionIdentitySigned(
      */
 
     private fun verifyPublicKeyLevelAndPurpose(publicKey: IdentityPublicKey) {
+        // If state transition requires MASTER security level it must be sign only with MASTER key
+        if (publicKey.isMaster() && getKeySecurityLevelRequirement() != IdentityPublicKey.SecurityLevel.MASTER) {
+            throw InvalidSignaturePublicKeySecurityLevelException(
+                IdentityPublicKey.SecurityLevel.MASTER,
+                this.getKeySecurityLevelRequirement(),
+            )
+        }
+
+        // Otherwise, key security level should be less than MASTER but more or equal than required
         if (getKeySecurityLevelRequirement() < publicKey.securityLevel) {
             throw PublicKeySecurityLevelNotMetException(
                 publicKey.securityLevel,
                 this.getKeySecurityLevelRequirement(),
             )
         }
 
-        if (publicKey.purpose !== IdentityPublicKey.Purpose.AUTHENTICATION) {
+        if (publicKey.purpose != IdentityPublicKey.Purpose.AUTHENTICATION) {
             throw WrongPublicKeyPurposeException(
                 publicKey.purpose,
                 IdentityPublicKey.Purpose.AUTHENTICATION,
             )
         }
     }
 
+    private fun verifyPublicKeyIsEnabled(publicKey: IdentityPublicKey) {
+        if (publicKey.disabledAt != null) {
+            throw PublicKeyIsDisabledException(publicKey)
+        }
+    }
+
     /**
      * Returns minimal key security level that can be used to sign this ST.
      * Override this method if the ST requires a different security level.
      *
      * @return {number}
      */
     open fun getKeySecurityLevelRequirement(): IdentityPublicKey.SecurityLevel {
-        return IdentityPublicKey.SecurityLevel.MASTER
+        return IdentityPublicKey.SecurityLevel.HIGH
```

Comment:
This was a big change in 0.23

Reply:
Identities will need a key with `HIGH` to sign state transitions (to create usernames, profiles, etc)